src/services/EmailApiContext.ts

```diff
@@ -0,0 +1,130 @@
+     * @returns The HTML content of the latest email.
+     */
+    async getEmailBody(): Promise<string> {
+      const response = await this.context.get('view/latest.html');
+      const buffer = await response.body();
+      const htmlString = buffer.toString('utf-8');
+      return htmlString;
+    }
+  
+    /**
+     * Generates the full email content, combining headers and body.
+     * @param email - The email address to fetch headers for.
+     * @returns The full email content as a string.
+     */
+    async generateEmailContent(email: string): Promise<string> {
+      const headers = await this.getEmailHeaders(email);
+      const htmlTemplate = await this.getEmailBody();
+      
+      const headerSection = `
+      <div style="font-family:arial; font-size:14px;">
+        <p>
+            From: ${headers.fromName} &lt;${headers.fromAddress}&gt;<br/><br/>
+            To: ${headers.toName} &lt;${headers.toAddress}&gt;<br/><br/>
+            Subject: ${headers.subject}<br/><br/>
+        </p>
+      </div>  
+    `;
+      const emailContent = headerSection + htmlTemplate;
+      return emailContent;
+    }
+  
+    /**
+     * Retrieves the plain text content of the latest email.
+     * @returns The plain text content of the latest email.
+     */
+    async getRenderMessageTxt(): Promise<string> {
+      const response = await this.context.get('view/latest.txt');
+      const buffer = await response.body();
+      const text = buffer.toString('utf-8');
+      return text;
+    }
+  
+    /**
+     * Extracts the first URL found in the plain text content of the latest email.
+     * @returns The first URL found in the email content.
+     * @throws An error if no URL is found in the email content.
+     */
+    async getLinkFromMail(): Promise<string> {
+      const textContent = await this.getRenderMessageTxt();
+      const urlMatch = textContent.match(/https?:\/\/[^\s]+/);
+      if (urlMatch && urlMatch.length > 0) {
+          return urlMatch[0];
+      }
+      throw new Error('No URL found in the email content');
+    }
+  
+    /**
+     * Deletes a specific email by ID if provided, or deletes all emails if no ID is provided.
+     * @param emailId - The ID of the email to delete (optional).
+     */
+    async deleteMail(emailId?: string): Promise<void> {
+      const data = emailId ? { IDs: [emailId] } : {};
+      await this.context.delete(`api/v1/messages`, { data });
+    }
+
+    /**
+     * Creates a new EmailApiContext instance with the appropriate configuration.
+     * @returns A promise that resolves to an EmailApiContext instance.
+     */
+    public static async create(): Promise<EmailApiContext> {
+        const extraHTTPHeaders: HTTPHeaders = {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+        };
+
+        const context = await request.newContext({
+            baseURL: 'http://localhost:8025/',
```

Comment:
we should not hard code the url, it might differ between local and ci

Reply:
Removed hardcoded url